feature-home/src/main/java/com/vanced/manager/feature/home/domain/repository/AppRepository.kt

```diff
@@ -0,0 +1,22 @@
+package com.vanced.manager.feature.home.domain.repository
+
+import com.vanced.manager.feature.home.domain.entity.App
+
+interface AppRepository {
+
+    suspend fun getMicroGInformation(
+        packageName: (App.MicroG.Companion) -> String
+    ): App.MicroG
+
+    suspend fun getVancedManagerInformation(
+        packageName: (App.VancedManager.Companion) -> String
```

Comment:
I suspect why you use higher order functions instead of a regular parameter (probably because of a try/catch block that's needed), but looks a bit weird